crit/mempages.go

```diff
@@ -193,3 +195,92 @@ func (mr *MemoryReader) GetShmemSize() (int64, error) {
 
 	return size, nil
 }
+
+// PatternMatch represents a match when searching for a pattern in memory.
+type PatternMatch struct {
+	Vaddr   uint64
+	Length  int
+	Context int
+	Pattern string
```

Comment:
`Pattern` is slightly misleading here. That would be the string supplied by the user - I feel this would be better named as `Result` or `Match`.